lib/spack/spack/cmd/load.py

```diff
@@ -108,16 +110,14 @@ def load(parser, args):
         )
         return 1
 
-    with spack.store.STORE.db.read_transaction():
-        if "dependencies" in args.things_to_load:
-            include_roots = "package" in args.things_to_load
-            specs = [
-                dep for spec in specs for dep in spec.traverse(root=include_roots, order="post")
-            ]
+    if args.things_to_load != "package,dependencies":
+        tty.warn(
+            "The `--only` flag in spack load is deprecated and will be removed in Spack v0.22"
+        )
```

Comment:
I think we should undo this, but in a follow-on before 0.21.  I don't think the semantics of `--only dependencies` are unclear, and I think people actually do use that, e.g. with `BundlePackages`.

Can you say what's unclear about `--only package`?  I don't think it's *that* unclear... I also think it's not great that it is incomplete... I'm not sure what you'd use it for.